src/datatype.c

```diff
@@ -360,19 +360,26 @@ void jl_compute_field_offsets(jl_datatype_t *st)
         // if we have no fields, we can trivially skip the rest
         if (st == jl_symbol_type || st == jl_string_type) {
             // opaque layout - heap-allocated blob
-            static const jl_datatype_layout_t opaque_byte_layout = {0, 1, -1, 1, 0, 0};
```

Comment:
```suggestion
static const struct {jl_datatype_layout_t fields; int8_t pointers} opaque_byte_layout = {{0, 1, -1, 1, 0, 0}, -1};
st->layout = &opaque_byte_layout.fields;
```

But seems like the reading this field would imply a problem is going to happen somewhere else shortly thereafter?